Engine/DataFeeds/Enumerators/FillForwardEnumerator.cs

```diff
@@ -300,7 +300,8 @@ protected virtual bool RequiresFillForwardData(TimeSpan fillForwardResolution, B
                 // it means that we have to calculate potentialBadEndTime in the same TZ
                 // by adding interval in daily resolution we move to next day
                 // in order to track daylight movement need to round-down origin day (before + Interval)
-                var potentialBarEndTime = RoundDown(item.ReferenceDateTime, item.Interval) + item.Interval;
+                var potentialBarEndTime = (RoundDown(item.ReferenceDateTime, item.Interval).ConvertTo(Exchange.TimeZone, _dataTimeZone) + item.Interval)
```

Comment:
Believe we should update the comments above and the _why_ we are doing these operations